src/test/java/core/basesyntax/MyCalculatorTest.java

```diff
@@ -0,0 +1,257 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class MyCalculatorTest {
+    private static final double DELTA = 0.0001;
+    private static MyCalculator myCalculator;
+
+    @BeforeAll
+    static void beforeAll() {
+        myCalculator = new MyCalculator();
+    }
+
+    @Test
+    void calculate_divisionByZero_notOk() {
+        assertThrows(MyCalculateException.class, () -> myCalculator.calculate(2.0, 0, '/'));
+    }
+
+    @Test
+    void calculate_singIsNotCorrect_notOk() {
+        assertThrows(MyCalculateException.class, () -> myCalculator.calculate(1, 1, '$'));
+    }
+
+    @Test
+    void calculate_additionTwoPositiveValues_Ok() throws Exception {
+        double actual = myCalculator.calculate(7.7, 3.3, '+');
+        double expected = 11;
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    void calculate_additionPositiveAndNegativeValues_Ok() throws Exception {
+        double actual = myCalculator.calculate(2, -1.1, '+');
+        double expected = 0.9;
+        assertEquals(actual, expected, DELTA);
+    }
+
+    @Test
+    void calculate_additionTowNegativeValues_Ok() throws Exception {
+        double actual = myCalculator.calculate(-2, -1.1, '+');
+        double expected = -3.1;
+        assertEquals(actual, expected, DELTA);
+    }
+
+    @Test
+    void calculate_additionWithFirstZero_Ok() throws Exception {
+        double actual = myCalculator.calculate(0, 7, '+');
+        double expected = 7;
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    void calculate_additionWithSecondZero_Ok() throws Exception {
+        double actual = myCalculator.calculate(10, 0, '+');
+        double expected = 10;
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    void calculate_additionMaxValue_notOk() {
+        assertThrows(MyCalculateException.class,
+                () -> myCalculator.calculate(Double.MAX_VALUE, 0.1, '+'));
+    }
+
+    @Test
+    void calculate_additionMinValue_notOk() {
+        assertThrows(MyCalculateException.class,
+                () -> myCalculator.calculate(-Double.MAX_VALUE, -0.1, '+'));
+    }
+
+    @Test
+    void calculate_subtractionTwoPositiveValues_Ok() throws Exception {
+        double actual = myCalculator.calculate(7.7, 3.3, '-');
+        double expected = 4.4;
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    void calculate_subtractionPositiveAndNegativeValues_Ok() throws Exception {
+        double actual = myCalculator.calculate(2, -1.1, '-');
+        double expected = 3.1;
+        assertEquals(actual, expected, DELTA);
+    }
+
+    @Test
+    void calculate_subtractionTowNegativeValues_Ok() throws Exception {
+        double actual = myCalculator.calculate((-2), -1.1, '-');
```

Comment:
```suggestion
        double actual = myCalculator.calculate(-2, -1.1, '-');
```
redundant parentheses

Reply:
+